hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -4937,11 +4937,11 @@ private CheckAndMutateResult checkAndMutateInternal(CheckAndMutate checkAndMutat
             boolean valueIsNull =
               comparator.getValue() == null || comparator.getValue().length == 0;
             if (result.isEmpty() && valueIsNull) {
-              matches = true;
-            } else if (result.size() > 0 && result.get(0).getValueLength() == 0 && valueIsNull) {
-              matches = true;
+              matches = op != CompareOperator.NOT_EQUAL;
```

Comment:
Oh! My comment has been pending here for many days! I forgot to submit it!

Reply:
Oh, is this documented any where?

Reply:
@sunhelly Please take a look at the above comment?

Reply:
@Apache9 Haven't documented this. But I have discussed this problem with @GeorryHuang on the wechat before he found the comment pending. 
The explanation here is that, NULL check is for equal or not equals, rare people use conditions like >null or <null. So if here we add LESS and GREATER, I think it may bring confusions.

Reply:
As @sunhelly  said.

Reply:
Then let's add this to the release note of this issue?

Reply:
I'll add it, thanks.